### PR TITLE
langchain[minor]: Make EmbeddingsFilters async

### PR DESCRIPTION
**Pull request by pprados**
Thank you for contributing to LangChain!

- [ ] **PR title**: community: EmbeddingsFilters is not compatible with async

- [ ] **PR message**: 
    - **Description:** EmbeddingsFilters is not compatible with async
    - **Issue:** Current implementation call sync methods
    - **Twitter handle:** pprados

- [X] **Add tests and docs**: Test were updated

- [X] **Lint and test**: Run `make format`, `make lint` and `make test` from the root of the package(s) you've modified. See contribution guidelines for more: https://python.langchain.com/docs/contributing/

@eyurtsev, another review ?

### CODE REVIEW

**Comment by pprados** [comment]
@eyurtsev 
Once again, an async bug

File: libs/community/tests/integration_tests/retrievers/test_contextual_compression.py [hunk 1]
```diff
@@ -24,3 +25,25 @@ def test_contextual_compression_retriever_get_relevant_docs() -> None:
     actual = retriever.invoke("Tell me about the Celtics")
     assert len(actual) == 2
     assert texts[-1] not in [d.page_content for d in actual]
+
+
+@pytest.mark.asyncio
```
Comment (eyurtsev) [inline libs/community/tests/integration_tests/retrievers/test_contextual_compression.py:30]:
```suggestion
```

File: libs/community/tests/integration_tests/retrievers/document_compressors/test_embeddings_filter.py [hunk 2]
```diff
@@ -41,3 +57,24 @@ def test_embeddings_filter_with_state() -> None:
     actual = relevant_filter.compress_documents(docs, query)
     assert len(actual) == 1
     assert texts[-1] == actual[0].page_content
+
+
+@pytest.mark.asyncio
```
Comment (eyurtsev) [inline libs/community/tests/integration_tests/retrievers/document_compressors/test_embeddings_filter.py:62]:
```suggestion
```

File: libs/community/tests/integration_tests/retrievers/document_compressors/test_embeddings_filter.py [hunk 3]
```diff
@@ -23,6 +24,21 @@ def test_embeddings_filter() -> None:
     assert len(set(texts[:2]).intersection([d.page_content for d in actual])) == 2
 
 
+@pytest.mark.asyncio
```
Comment (eyurtsev) [inline libs/community/tests/integration_tests/retrievers/document_compressors/test_embeddings_filter.py:27]:
```suggestion
```

File: libs/community/tests/integration_tests/retrievers/document_compressors/test_embeddings_filter.py [hunk 4]
```diff
@@ -1,5 +1,6 @@
 """Integration test for embedding-based relevant doc filtering."""
 import numpy as np
+import pytest
```
Comment (eyurtsev) [inline libs/community/tests/integration_tests/retrievers/document_compressors/test_embeddings_filter.py:3]:
```suggestion
```

File: libs/community/tests/integration_tests/retrievers/document_compressors/test_base.py [hunk 5]
```diff
@@ -27,3 +28,25 @@ def test_document_compressor_pipeline() -> None:
     actual = pipeline_filter.compress_documents(docs, "Tell me about farm animals")
     assert len(actual) == 1
     assert actual[0].page_content in texts[:2]
+
+
+@pytest.mark.asyncio
```
Comment (eyurtsev) [inline libs/community/tests/integration_tests/retrievers/document_compressors/test_base.py:33]:
```suggestion
```

File: libs/community/tests/integration_tests/retrievers/document_compressors/test_base.py [hunk 6]
```diff
@@ -1,4 +1,5 @@
 """Integration test for compression pipelines."""
+import pytest
```
Comment (eyurtsev) [inline libs/community/tests/integration_tests/retrievers/document_compressors/test_base.py:2]:
```suggestion
```

**Review by eyurtsev — COMMENTED** [review]
Looks good -- could you undo changes in pyproject.toml and poetry.lock -- most PRs should leave those unomdified

File: libs/community/tests/integration_tests/retrievers/document_compressors/test_embeddings_filter.py (hunk 2 above)
Comment (pprados) [reply libs/community/tests/integration_tests/retrievers/document_compressors/test_embeddings_filter.py:62]: if I delete this annotation, the test is not run.

File: libs/community/tests/integration_tests/retrievers/document_compressors/test_base.py (hunk 5 above)
Comment (pprados) [reply libs/community/tests/integration_tests/retrievers/document_compressors/test_base.py:33]: if I delete this annotation, the test is not run

File: libs/community/tests/integration_tests/retrievers/document_compressors/test_base.py (hunk 6 above)
Comment (pprados) [reply libs/community/tests/integration_tests/retrievers/document_compressors/test_base.py:2]: if I delete this annotation, the test is not run

File: libs/community/tests/integration_tests/retrievers/document_compressors/test_embeddings_filter.py (hunk 4 above)
Comment (pprados) [reply libs/community/tests/integration_tests/retrievers/document_compressors/test_embeddings_filter.py:3]: if I delete this annotation, the test is not run

Comment (eyurtsev) [reply libs/community/tests/integration_tests/retrievers/document_compressors/test_embeddings_filter.py:3]: Make sure your environment is up to date. We use a plugin to automatically run asyncio code. 

See:
https://github.com/langchain-ai/langchain/blob/5ba1899cd72dd57ee760d63137e2c92731c7f7a4/libs/langchain/pyproject.toml#L145-L145

Comment (eyurtsev) [reply libs/community/tests/integration_tests/retrievers/document_compressors/test_embeddings_filter.py:3]: https://pytest-asyncio.readthedocs.io/en/latest/concepts.html#auto-mode

File: libs/community/tests/integration_tests/retrievers/test_contextual_compression.py (hunk 1 above)
Comment (pprados) [reply libs/community/tests/integration_tests/retrievers/test_contextual_compression.py:30]: if I delete this annotation, the test is not run.

File: libs/community/tests/integration_tests/retrievers/document_compressors/test_embeddings_filter.py (hunk 2 above)
Comment (pprados) [reply libs/community/tests/integration_tests/retrievers/document_compressors/test_embeddings_filter.py:62]: if I delete this annotation, the test is not run.

File: libs/community/tests/integration_tests/retrievers/document_compressors/test_embeddings_filter.py (hunk 3 above)
Comment (pprados) [reply libs/community/tests/integration_tests/retrievers/document_compressors/test_embeddings_filter.py:27]: if I delete this annotation, the test is not run.

**Comment by pprados** [comment]
Hello @eyurtsev 

In test_embeddings_filter.py, I change
`from langchain_community.embeddings import OpenAIEmbeddings`
to
`from langchain_openai.embeddings import OpenAIEmbeddings`

But, for that, I need to add a dependency in pyproject.tml
```toml
[tool.poetry.group.test_integration.dependencies]
...
langchain-openai = ">=0.1.8"
```

But, the `poetry lock --no-update` return:
```
Because langchain-openai (0.1.8) depends on tiktoken (>=0.7,<1)
 and no versions of langchain-openai match >0.1.8, langchain-openai (>=0.1.8) requires tiktoken (>=0.7,<1).
So, because langchain-community depends on both tiktoken (>=0.3.2,<0.6.0) and langchain-openai (>=0.1.8), version solving failed.
```

Do you think I should keep the original import?

**Comment by eyurtsev** [comment]
> Hello @eyurtsev
> 
> In test_embeddings_filter.py, I change `from langchain_community.embeddings import OpenAIEmbeddings` to `from langchain_openai.embeddings import OpenAIEmbeddings`
> 
> But, for that, I need to add a dependency in pyproject.tml
> 
> ```toml
> [tool.poetry.group.test_integration.dependencies]
> ...
> langchain-openai = ">=0.1.8"
> ```
> 
> But, the `poetry lock --no-update` return:
> 
> ```
> Because langchain-openai (0.1.8) depends on tiktoken (>=0.7,<1)
>  and no versions of langchain-openai match >0.1.8, langchain-openai (>=0.1.8) requires tiktoken (>=0.7,<1).
> So, because langchain-community depends on both tiktoken (>=0.3.2,<0.6.0) and langchain-openai (>=0.1.8), version solving failed.
> ```
> 
> Do you think I should keep the original import?


Yes, please keep the original import.

If you want to spend more effort, you could fix the test logic itself (not required for this PR) -- the test should not depend on the embedding provider, and instead can leverage fake embeddings in langchain_core. If that's done, then the test can be moved into unit tests.

**Comment by pprados** [comment]
@eyurtsev 
"The Integration docs lint" checks is blocked in all the worlflow.

**Review by eyurtsev — APPROVED** [review]

**Pull request merged**